src/equations/shallow_water_2d.jl

```diff
@@ -177,24 +177,18 @@ For details see Section 9.2.5 of the book:
   1st edition
   ISBN 0471987662
 """
-@inline function boundary_condition_slip_wall(u_inner, normal_direction::AbstractVector,
+@inline function boundary_condition_slip_wall(u_inner, orientation_or_normal, direction,
```

Comment:
Changing this call signature is causing a test to fail. This should revert back to using `normal_direction::AbstractVector` and two additional routines should be added (basically borrowed from how we do this for the 2D compressible Euler equations).
```julia
"""
    boundary_condition_slip_wall(u_inner, orientation, direction, x, t,
                                 surface_flux_function, equations::ShallowWaterEquations2D)

Should be used together with [`TreeMesh`](@ref).
"""
@inline function boundary_condition_slip_wall(u_inner, orientation,
                                              direction, x, t,
                                              surface_flux_function,
                                              equations::ShallowWaterEquations2D)
  # get the appropriate normal vector from the orientation
  if orientation == 1
    normal_direction = SVector(1, 0)
  else # orientation == 2
    normal_direction = SVector(0, 1)
  end

  # compute and return the flux using `boundary_condition_slip_wall` routine above
  return boundary_condition_slip_wall(u_inner, normal_direction, direction,
                                      x, t, surface_flux_function, equations)
end

"""
    boundary_condition_slip_wall(u_inner, normal_direction, direction, x, t,
                                 surface_flux_function, equations::ShallowWaterEquations2D)

Should be used together with [`StructuredMesh`](@ref).
"""
@inline function boundary_condition_slip_wall(u_inner, normal_direction::AbstractVector,
                                              direction, x, t,
                                              surface_flux_function,
                                              equations::ShallowWaterEquations2D)
  # flip sign of normal to make it outward pointing, then flip the sign of the normal flux back
  # to be inward pointing on the -x and -y sides due to the orientation convention used by StructuredMesh
  if isodd(direction)
    boundary_flux = -boundary_condition_slip_wall(u_inner, -normal_direction,
                                                  x, t, surface_flux_function, equations)
  else
    boundary_flux = boundary_condition_slip_wall(u_inner, normal_direction,
                                                 x, t, surface_flux_function, equations)
  end

  return boundary_flux
end
```

Reply:
Ok, so I amjust add this to `src/equations/shallow_water_2d.jl` and have 3 `boundary_condition_slip_wall` functions in total?

Reply:
yes that all looks good now